nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -877,6 +878,120 @@ impl<'a> Planner<'a> {
         Ok(())
     }
 
+    /// Update at most one existing zone to use a new image source.
+    fn do_plan_zone_updates(&mut self) -> Result<(), Error> {
+        // We are only interested in non-decommissioned sleds.
+        let sleds = self
+            .input
+            .all_sleds(SledFilter::Commissioned)
+            .map(|(id, _details)| id)
+            .collect::<Vec<_>>();
+
+        // Wait for all current zones to appear in the inventory.
+        // It would be nice if we could check their image source,
+        // but the inventory doesn't report that correctly: see
+        // <https://github.com/oxidecomputer/omicron/issues/8084>.
+        let inventory_zones = self
+            .inventory
+            .all_omicron_zones()
+            .map(|z| (z.id, z))
+            .collect::<BTreeMap<_, _>>();
```

Comment:
Nit - could this be a `BTreeSet` of just the zone IDs? It looks like we don't care about the value. (I also wonder if we need to build this intermediate at all - could the loop below operate on `inventory.sled_agents` directly?)

Reply:
Yup, fixed in e31fc6098.